lib/compress.js

```diff
@@ -4655,4 +4655,36 @@ merge(Compressor.prototype, {
     OPT(AST_PrefixedTemplateString, function(self, compressor){
         return self;
     });
+
+    OPT(AST_ObjectKeyVal, function(self, compressor){
+        // p:function(){} ---> p(){}
+        // p:function*(){} ---> *p(){}
+        // p:async function(){} ---> async p(){}
+        // p:()=>{} ---> p(){}
+        // p:*()=>{} ---> *p(){}
+        // p:async()=>{} ---> async p(){}
+        if (compressor.option("ecma") >= 6) {
+            var key = self.key;
+            var value = self.value;
+            var is_arrow_with_block = value instanceof AST_Arrow && Array.isArray(value.body);
+            if ((is_arrow_with_block || value instanceof AST_Function)
+                && !value.name) {
+                return make_node(AST_ConciseMethod, self, {
+                    async: value.async,
+                    is_generator: value.is_generator,
+                    key: key instanceof AST_Node ? key : make_node(AST_SymbolMethod, self, {
+                        name: key,
+                    }),
+                    value: make_node(AST_Accessor, value, {
+                        name: null,
+                        argnames: value.argnames,
+                        async: value.async,
+                        is_generator: value.is_generator,
+                        body: value.body,
```

Comment:
Does using CLI against
```js
(function(){var quux=()=>{console.log("PASS")};return{prop(){console.log;quux()}}})().prop();
```
proceeds any further? If not, then may be `AST_ConciseMethod` is somehow blocking a transformation somewhere.